zcash_primitives/src/transaction/components/sapling/builder.rs

```diff
@@ -60,6 +60,13 @@ impl fmt::Display for Error {
     }
 }
 
+/// A trait that provides a minimized view of a Sapling input suitable for use in
+/// fee calculation.
+pub trait SaplingInput {
```

Comment:
Same comment here as above: since this is *only* relevant to tx under construction, and not anything in the protocol, I think it's better to keep this here and move `TzeInput` the other way.